### PR TITLE
Add GCP secrets

### PR DESCRIPTION
**Pull request by rhys-saldanha**

### CODE REVIEW

File: dist/src/main/resources/bootstrap-cloud.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+spring:
+  cloud:
+    gcp:
+      core:
+        enabled: true
+      secretmanager:
+        enabled: true
+
+DATABASE_USERNAME: ${sm://reciplease-database-username}
```
Comment (VikashKothary) [inline dist/src/main/resources/bootstrap-cloud.yml:9]: Maybe we can have best of both worlds: `${RECIPLEASE_DATABASE_USERNAME:sm://reciplease-database-username`?

Allows simulating production on our local machines if needed.

**Review by VikashKothary — APPROVED** [review]
Minor comment i.e. not a blocker.

Otherwise LGTM.

File: scripts/test-backend.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# description: Test maven and image build
+
+${MVN} package
+
+if [[ ! $? ]]; then
+  echo "Maven build failed"
+  exit 1
+fi
+
+make image
```
Comment (VikashKothary) [inline scripts/test-backend.sh]: Use ${MAKE}

File: dist/src/main/resources/bootstrap-cloud.yml (hunk 1 above)
Comment (rhys-saldanha) [reply dist/src/main/resources/bootstrap-cloud.yml:9]: local simulation still works! this bootstrap is an extension to the `db-prod` profile - locally you can run `db-prod` with your fancy ENV variables, then in GCP we run `cloud, db-prod`

File: Makefile [hunk 3]
```diff
@@ -36,7 +36,7 @@ init-backend:
 test: test-backend # alias for quick access
 test-backend:
 	@cd ${RECIPLEASE_PATH} && \
-	${MVN} test
+	./scripts/test-backend.sh
```
Comment (VikashKothary) [inline Makefile]: Normally this command is for running code related tests. i.e. for TDD. The checks for deployment should be in the actual deployment scripts.

File: scripts/test-backend.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+# file: test-backend.sh
+# description: Test maven and image build
```
Comment (VikashKothary) [inline scripts/test-backend.sh]: These checks should be a part of their respective commands i.e. 
[ ] create scripts/package-backend.sh
[ ] add to end of scripts/image-backend.sh

Currently your CI will run the scripts and process even if those commands fail.

**Review by VikashKothary — CHANGES_REQUESTED** [review]

File: dist/src/main/resources/bootstrap-cloud.yml (hunk 1 above)
Comment (VikashKothary) [reply dist/src/main/resources/bootstrap-cloud.yml:9]: Okay got it. So we'd only run `cloud` when it's running on GCP. That's fine. But for argument sake, I wanted to debug an PROD issue related to GCP. Could I use the profile `cloud` locally and everything will still work?

Regardless, until we've got an issue of that nature. It's not a major issue so I'd consider this point resolved.

File: Makefile (hunk 3 above)
Comment (rhys-saldanha) [reply Makefile]: hmm but then it doesn't run for PRs

File: dist/pom.xml [hunk 5]
```diff
@@ -64,7 +68,8 @@
                     <skip>false</skip>
                     <finalName>reciplease-dist</finalName>
                     <layers>
-                        <configuration>${project.basedir}/src/layers.xml</configuration>
+                        <enabled>true</enabled>
+                        <configuration>${projecgit t.basedir}/src/layers.xml</configuration>
```
Comment (VikashKothary) [inline dist/pom.xml]: Typo?

File: dist/src/main/resources/bootstrap-cloud.yml (hunk 1 above)
Comment (rhys-saldanha) [reply dist/src/main/resources/bootstrap-cloud.yml:9]: if you have the ENV variables for db details, you don't need `cloud`. if you want to run with gcloud features locally, there are ways of doing that, and if we run into GCO issues that we want to recreate locally, I can add a profile / make magic to set that up

**Review by VikashKothary — APPROVED** [review]

File: dist/pom.xml (hunk 5 above)
Comment (rhys-saldanha) [reply dist/pom.xml]: nice

File: dist/src/main/resources/bootstrap-cloud.yml (hunk 1 above)
Comment (rhys-saldanha) [reply dist/src/main/resources/bootstrap-cloud.yml:9]: there's an argument that we could remove the use of ENV variables and just use gcloud to store secrets... will require you to install the gcloud CLI

**Pull request merged**